functions/utils/index.js

```diff
@@ -1,9 +1,18 @@
+import { RelationKind } from './contants';
+
 export const now = () =>
   new Date().toISOString().slice(0, 19).replace('T', ' ');
 
 const isRecord = (value) =>
   value && typeof value === 'object' && !Array.isArray(value);
 
+const isCollection = (value) =>
+  Array.isArray(value) && (isRecord(value) || typeof value[0] === 'undefined');
```

Comment:
Can you explain why we need `&& (isRecord(value) || typeof value[0] === 'undefined');`?